soccer/src/soccer/control/motion_control.cpp

```diff
@@ -121,9 +128,23 @@ void MotionControl::run(const RobotState& state, const planning::Trajectory& tra
     // TODO(Kyle): Clamp acceleration
 
     Twist correction = Twist::zero();
+
     if (maybe_pose_target) {
         Pose error = maybe_pose_target.value() - state.pose;
         error.heading() = fix_angle_radians(error.heading());
+
+        std_msgs::msg::Float64 error_x_msg;
+        error_x_msg.data = error.position().x();
+        error_x_pub_->publish(error_x_msg);
```

Comment:
Check if there are subscribers before publishing. This way, we're not using resources if no one is looking at the graph.